### ISSUE TITLE
PointCloud.get_oriented_bounding_box() sometimes returns left-handed rotation matrix

### ISSUE THREAD

**Issue by jfelip** [isl-org/Open3D#2206]
**Describe the bug**
PointCloud.get_oriented_bounding_box() sometimes returns left-handed rotation matrix.

**To Reproduce**

This code example shows how sometimes the OBB R matrix is not right-handed. You might need to close the visualization window several times until the OBB matrix fails the right-handedness test.

```
import open3d as o3d
import numpy as np


def check_righthanded(R, verbose=False):
    res = True

    # Check that axis are right-handed
    if not np.allclose(R[2], np.cross(R[0], R[1])):
        if verbose:
            print("x cross y != z", np.cross(R[0], R[1]), " != ", R[2])
        res = False

    if not np.allclose(R[0], np.cross(R[1], R[2])):
        if verbose:
            print("y cross z != x", np.cross(R[1], R[2]), " != ", R[0])
        res = False

    if not np.allclose(R[1], np.cross(R[2], R[0])):
        if verbose:
            print("z cross x != y", np.cross(R[2], R[0]), " != ", R[1])
        res = False

    # Check that axis' are right-handed
    if not np.allclose(R[:, 2], np.cross(R[:, 0], R[:, 1])):
        if verbose:
            print("x' cross y' != z'", np.cross(R[:, 0], R[:, 1]), " != ", R[:, 2])
        res = False

    if not np.allclose(R[:, 0], np.cross(R[:, 1], R[:, 2])):
        if verbose:
            print("y' cross z' != x'", np.cross(R[:, 1], R[:, 2]), " != ", R[:, 0])
        res = False

    if not np.allclose(R[:, 1], np.cross(R[:, 2], R[:, 0])):
        if verbose:
            print("z' cross x' != y'", np.cross(R[:, 2], R[:, 0]), " != ", R[:, 1])
        res = False

    return res

while True:
    # Create point cloud data
    points = np.random.uniform(np.array([0, 0, 0]), np.array([.1, .2, 0.01]), [1000,3])
    pcd = o3d.geometry.PointCloud(points=o3d.utility.Vector3dVector(points))
    pcd.rotate(o3d.geometry.get_rotation_matrix_from_xyz([0, 0, .3]), center=np.array([.1, .5, 0.01])/2)

    # Get point cloud oriented bounding box
    obb = pcd.get_oriented_bounding_box()

    # Check if the OBB rotation matrix is right-handed
    R = np.copy(obb.R)
    if not check_righthanded(R, verbose=True):
        R[:, 2] = R[:, 2] * -1

    # Add the bounding box local reference frame
    obb_frame = o3d.geometry.TriangleMesh.create_coordinate_frame(size=.1)
    obb_frame.translate(obb.center)
    obb_frame.rotate(obb.R, center=obb.center)

    # Draw a box that represents the oriented BB
    obb2 = o3d.geometry.TriangleMesh.create_box(obb.extent[0],obb.extent[1],obb.extent[2])
    # Rotate to align the box to the target oriented bounding box
    obb2.rotate(obb.R, center=obb.extent/2)
    # Move it to the center, the OBB has its center at the middle of all vertices while the box has the center
    # at the lower left corner. This translation is needed to account for that.
    obb2.translate(obb.center, relative=False)

    # Move the point cloud up to be able to see it above the fitted OBB
    pcd.translate([0, 0, 0.01], relative=True)

    o3d.visualization.draw_geometries([pcd, obb, obb_frame, obb2,
                                       o3d.geometry.TriangleMesh.create_coordinate_frame(size=.1)])

```

**Expected behavior**
The rotation matrix obtained from the OBB should be right-handed. In the code above there is a solution proposed if the right-handedness test fails, which is to flip the z-axis.
```
    if not check_righthanded(R, verbose=True):
        R[:, 2] = R[:, 2] * -1
```

**Screenshots**
An example of the effect of rotating an object with a left-handed matrix. It seems to flip the face normals, this can be seen in the box and its reference frame which are showing the inner triangles.
![image](https://user-images.githubusercontent.com/6513981/90700120-d8691500-e239-11ea-996d-cef2c5c630c7.png)

**Environment (please complete the following information):**

- Operating system: Ubuntu 18.04
- Python version: (e.g. Python 3.6)
- Open3D version: 0.10.0
- Is this remote workstation?: no
- How did you install Open3D?: pip

**Comment by xXBasti** [comment]
I am facing the same Issue, is there an update?

**Comment by shinxg** [comment]
> **Describe the bug**
> PointCloud.get_oriented_bounding_box() sometimes returns left-handed rotation matrix.
> 
> **To Reproduce**
> 
> This code example shows how sometimes the OBB R matrix is not right-handed. You might need to close the visualization window several times until the OBB matrix fails the right-handedness test.
> 
> ```
> import open3d as o3d
> import numpy as np
> 
> 
> def check_righthanded(R, verbose=False):
>     res = True
> 
>     # Check that axis are right-handed
>     if not np.allclose(R[2], np.cross(R[0], R[1])):
>         if verbose:
>             print("x cross y != z", np.cross(R[0], R[1]), " != ", R[2])
>         res = False
> 
>     if not np.allclose(R[0], np.cross(R[1], R[2])):
>         if verbose:
>             print("y cross z != x", np.cross(R[1], R[2]), " != ", R[0])
>         res = False
> 
>     if not np.allclose(R[1], np.cross(R[2], R[0])):
>         if verbose:
>             print("z cross x != y", np.cross(R[2], R[0]), " != ", R[1])
>         res = False
> 
>     # Check that axis' are right-handed
>     if not np.allclose(R[:, 2], np.cross(R[:, 0], R[:, 1])):
>         if verbose:
>             print("x' cross y' != z'", np.cross(R[:, 0], R[:, 1]), " != ", R[:, 2])
>         res = False
> 
>     if not np.allclose(R[:, 0], np.cross(R[:, 1], R[:, 2])):
>         if verbose:
>             print("y' cross z' != x'", np.cross(R[:, 1], R[:, 2]), " != ", R[:, 0])
>         res = False
> 
>     if not np.allclose(R[:, 1], np.cross(R[:, 2], R[:, 0])):
>         if verbose:
>             print("z' cross x' != y'", np.cross(R[:, 2], R[:, 0]), " != ", R[:, 1])
>         res = False
> 
>     return res
> 
> while True:
>     # Create point cloud data
>     points = np.random.uniform(np.array([0, 0, 0]), np.array([.1, .2, 0.01]), [1000,3])
>     pcd = o3d.geometry.PointCloud(points=o3d.utility.Vector3dVector(points))
>     pcd.rotate(o3d.geometry.get_rotation_matrix_from_xyz([0, 0, .3]), center=np.array([.1, .5, 0.01])/2)
> 
>     # Get point cloud oriented bounding box
>     obb = pcd.get_oriented_bounding_box()
> 
>     # Check if the OBB rotation matrix is right-handed
>     R = np.copy(obb.R)
>     if not check_righthanded(R, verbose=True):
>         R[:, 2] = R[:, 2] * -1
> 
>     # Add the bounding box local reference frame
>     obb_frame = o3d.geometry.TriangleMesh.create_coordinate_frame(size=.1)
>     obb_frame.translate(obb.center)
>     obb_frame.rotate(obb.R, center=obb.center)
> 
>     # Draw a box that represents the oriented BB
>     obb2 = o3d.geometry.TriangleMesh.create_box(obb.extent[0],obb.extent[1],obb.extent[2])
>     # Rotate to align the box to the target oriented bounding box
>     obb2.rotate(obb.R, center=obb.extent/2)
>     # Move it to the center, the OBB has its center at the middle of all vertices while the box has the center
>     # at the lower left corner. This translation is needed to account for that.
>     obb2.translate(obb.center, relative=False)
> 
>     # Move the point cloud up to be able to see it above the fitted OBB
>     pcd.translate([0, 0, 0.01], relative=True)
> 
>     o3d.visualization.draw_geometries([pcd, obb, obb_frame, obb2,
>                                        o3d.geometry.TriangleMesh.create_coordinate_frame(size=.1)])
> ```
> 
> **Expected behavior**
> The rotation matrix obtained from the OBB should be right-handed. In the code above there is a solution proposed if the right-handedness test fails, which is to flip the z-axis.
> 
> ```
>     if not check_righthanded(R, verbose=True):
>         R[:, 2] = R[:, 2] * -1
> ```
> 
> **Screenshots**
> An example of the effect of rotating an object with a left-handed matrix. It seems to flip the face normals, this can be seen in the box and its reference frame which are showing the inner triangles.
> ![image](https://user-images.githubusercontent.com/6513981/90700120-d8691500-e239-11ea-996d-cef2c5c630c7.png)
> 
> **Environment (please complete the following information):**
> 
> * Operating system: Ubuntu 18.04
> * Python version: (e.g. Python 3.6)
> * Open3D version: 0.10.0
> * Is this remote workstation?: no
> * How did you install Open3D?: pip

Thanks a lot. Nice solution! Hope this bug will be fixed soon.

**Comment by benjaminum** [comment]
Thanks for reporting. Fixed in #4452
